desktop/core/src/desktop/api2_tests.py

```diff
@@ -726,6 +726,9 @@ def test_gist_directory_creation(self):
 
     assert_true(home_dir.children.filter(name='Gist').exists())
 
+    gist_dir2 = Directory.objects.create(name='Gist', owner=self.user, parent_directory=home_dir)
+    assert_raises(Directory.MultipleObjectsReturned, Document2.objects.get_gist_directory, self.user)
```

Comment:
https://github.com/cloudera/hue/issues/1809#issuecomment-784408951 is what we want to repro..

Reply:
Updated the test_create UT along with the code fix @romainr